pkgs/by-name/xa/xar/package.nix

```diff
@@ -0,0 +1,175 @@
+  # picked up by stdenv’s C compiler wrapper (see ccWrapper_addCVars). This
+  # doesn’t really belong here and either should be part of libxml2 package or
+  # libxml2 in Nixpkgs can just fix their header paths.
+  env.NIX_CFLAGS_COMPILE = "-isystem ${libxml2.dev}/include/libxml2";
+
+  buildInputs =
+    [
+      zlib
+      libxml2
+    ]
+    # Darwin uses CommonCrypto, other platforms use openssl.
+    ++ lib.optional stdenv.hostPlatform.isDarwin darwin.CommonCrypto
+    ++ lib.optional (!stdenv.hostPlatform.isDarwin) openssl
+    ++ lib.optional withExt2 e2fsprogs
+    ++ lib.optional withAcl acl
+    ++ lib.optional withBzip2 bzip2
+    ++ lib.optional withLzma xz
+    ++ lib.optional stdenv.hostPlatform.isMusl musl-fts;
+
+  passthru.tests = {
+    version = testers.testVersion {
+      package = xar;
+      version = "1.8dev";
+    };
+
+    # Check that xar builds without optional dependencies.
+    minimal = xar.override {
+      withExt2 = false;
+      withAcl = false;
+      withBzip2 = false;
+      withLzma = false;
+    };
+
+    integrationTest =
+      let
+        python = pythonForTests python3.pythonOnBuildForHost;
+      in
+      runCommand "xar-tests-integration-test"
+        {
+          inherit src;
+          python = python.executable;
+          nativeBuildInputs = [
+            xar
+            python
+            libxslt
+          ];
+        }
+        ''
+          "$python" -- "$src"/xar/test/run-all.py
+          touch -- "$out"
+        '';
+
+    smokeTest =
+      runCommand "xar-tests-smoke-test"
+        {
+          inherit src;
+          depsBuildBuild = [
+            buildPackages.stdenv.cc
+            xar
+            openssl
+          ];
+        }
+        ''
+          cp -- "$src"/xar/test/{buffer.c,validate.c} .
+          "$CC_FOR_BUILD" -lxar -o buffer buffer.c
+          "$CC_FOR_BUILD" -lxar -lcrypto -o validate validate.c
+          ./buffer validate.c
+          xar -x -f test.xar
+          diff validate.c mydir/secondfile
+          ./validate test.xar
+          touch -- "$out"
+        '';
+
+    # For convenience, we expose manualTest to allow testing xar manually (since
+    # extended attributes are not supported in Nix sandbox).
+    #
+    # Run with
+    # $ manualTest=$(nix build --no-link --print-out-paths --file . xar.tests.manualTest) && "$manualTest"
+    # Ensure that all tests are PASSED and none are FAILED or SKIPPED.
+    manualTest =
+      runCommand "xar-manual-test"
+        {
+          inherit src xar libxslt;
+          python = (pythonForTests python3).interpreter;
+          nativeBuildInputs = [ makeWrapper ];
+        }
+        ''
+          makeWrapper "$python" "$out" \
+            --prefix PATH : "$xar/bin" \
+            --suffix PATH : "$libxslt/bin" \
+            --add-flags -- \
+            --add-flags "$src/xar/test/run-all.py"
+        '';
+  };
+
+  passthru.updateScript = gitUpdater { };
```

Comment:
How is this going to work when we’re `fetchurl`ing a tarball?

Reply:
`meta.homepage` takes precedence over `src`.

Interface:
https://github.com/NixOS/nixpkgs/blob/ab16e260852059740a6ee5d87a506cf918735bc1/pkgs/common-updater/git-updater.nix#L14-L16

Implementation:
https://github.com/NixOS/nixpkgs/blob/164d4463dfeda1c15594e72e60a2c7e80962aa7c/pkgs/common-updater/scripts/list-git-tags#L41-L46

Not sure if gitUpdater is documented in Nixpkgs manual though, ^F didn’t give any results (but perhaps I was looking in the wrong place).

That said, thanks for pointing that out, I’ve added `updateScript` and completely forgot to test that it actually works. Probably needs `rev-prefix` because tags are prefixed with `xar-`.

Reply:
It uses `src.meta.homepage`, which is different to `meta.homepage` – that’s how `fetchFromGitHub` etc. expose their repository URLs. So I don’t think this will actually work, unless I’m misunderstanding something.

What do you think about using `fetchFromGitHub` and `nix-update-script` instead? That would use the GitHub API directly for updates and also ensure that the sources match what’s in the repository.

Reply:
NB using `nix-update-script` is blocked by https://github.com/Mic92/nix-update/pull/269 that is not released yet.

Reply:
The update script should work once Nixpkgs contains the next nix-update release. Tested after manually bumping `nix-update` to unreleased commit and rolling back `xar` to a previous release.

<details>

```diff
diff --git a/pkgs/by-name/ni/nix-update/package.nix b/pkgs/by-name/ni/nix-update/package.nix
index db47d4b5abd6..21f53318d0da 100644
--- a/pkgs/by-name/ni/nix-update/package.nix
+++ b/pkgs/by-name/ni/nix-update/package.nix
@@ -9,14 +9,14 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "nix-update";
-  version = "1.5.0";
+  version = "unstable-2024-08-21";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "Mic92";
     repo = pname;
-    rev = version;
-    hash = "sha256-zWkJs6anDddTiUestnqGVfG1GErLJr5EarWI3DcKZpU=";
+    rev = "737121eccb67542e8c004c64da833fede2e80c64";
+    hash = "sha256-xn0dC4M3mfItxP+s3/v3Hz/CSKp74VH/gMfufKxl9/4=";
   };
 
   nativeBuildInputs = [
diff --git a/pkgs/by-name/xa/xar/package.nix b/pkgs/by-name/xa/xar/package.nix
index 4ad071f36a1f..c77b75b3c09c 100644
--- a/pkgs/by-name/xa/xar/package.nix
+++ b/pkgs/by-name/xa/xar/package.nix
@@ -31,13 +31,13 @@
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "xar";
-  version = "498";
+  version = "496";
 
   src = fetchFromGitHub {
     owner = "apple-oss-distributions";
     repo = "xar";
     rev = "xar-${finalAttrs.version}";
-    hash = "sha256-RyWeR/ZnDBHIZhwzVxETdrTTPQA2VgsLZegRkxX1240=";
+    hash = "sha256-MpV93JstGuoKJJWhG3Tckv1q2N3mTVg60LSDAGqoCOQ=";
   };
 
   # Update patch set with
```
</details>

Running `nix develop --file maintainers/scripts/update.nix --argstr package xar` bumps the `xar` version and updates `hash` as expected.

Reply:
Marking this thread as resolved based on the comment above since nix-update release is non-blocking for the package implementation.